src/bp-document/bp-document-functions.php

```diff
@@ -3761,3 +3449,1309 @@ function bb_document_user_can_upload( $user_id = 0, $group_id = 0 ) {
+	}
+	return $image_editors;
+}
+
+/**
+ * Remove all temp directory.
+ *
+ * @param string $dir directory to remove.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bp_document_remove_temp_directory( $dir ) {
+	if ( is_dir( $dir ) ) {
+		$objects = scandir( $dir );
+		foreach ( $objects as $object ) {
+			if ( '.' !== $object && '..' !== $object ) {
+				if ( filetype( $dir . '/' . $object ) === 'dir' ) {
+					bp_document_remove_temp_directory( $dir . '/' . $object );
+				} else {
+					unlink( $dir . '/' . $object );
+				}
+			}
+		}
+		reset( $objects );
+		rmdir( $dir );
+	}
+}
+
+/**
+ * Generate the document preview.
+ *
+ * @param int $attachment_id attachment id.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bp_document_generate_document_previews( $attachment_id ) {
+
+	$is_preview_generated = get_post_meta( $attachment_id, 'document_preview_generated', true );
+
+	if ( empty( $is_preview_generated ) ) {
+		$extension = bp_document_extension( $attachment_id );
+		if ( in_array( $extension, bp_get_document_preview_code_extensions(), true ) ) {
+			bp_document_generate_code_previews( $attachment_id );
+		}
+	}
+
+	$extension                   = bp_document_extension( $attachment_id );
+	$activity_file               = image_get_intermediate_size( $attachment_id, 'bb-document-pdf-preview-activity-image' );
+	$image_pdf_popup_file        = image_get_intermediate_size( $attachment_id, 'bb-document-pdf-image-popup-image' );
+	$preview_image_activity_file = image_get_intermediate_size( $attachment_id, 'bb-document-image-preview-activity-image' );
+
+	if ( 'pdf' === $extension && ( ! $activity_file || ! $image_pdf_popup_file || ! $preview_image_activity_file ) ) {
+
+		// Add upload filters.
+		bb_document_add_upload_filters();
+
+		add_filter( 'wp_image_editors', 'bp_document_wp_image_editors' );
+
+		// Register fallback intermediate image sizes.
+		bb_document_add_fallback_intermediate_image_sizes();
+
+		// Register image sizes.
+		bb_document_register_image_sizes();
+
+		bp_document_pdf_previews( array( $attachment_id ), true );
+
+		// Deregister image sizes.
+		bb_document_deregister_image_sizes();
+
+		// Register fallback intermediate image sizes.
+		bb_document_remove_fallback_intermediate_image_sizes();
+
+		remove_filter( 'wp_image_editors', 'bp_document_wp_image_editors' );
+
+		// Remove upload filters.
+		bb_document_remove_upload_filters();
+	}
+}
+
+/**
+ * Helper to set the max_execution_time.
+ *
+ * @param int $time_limit Time limit.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bp_document_set_time_limits( $time_limit ) {
+	$max_execution_time = ini_get( 'max_execution_time' );
+	if ( $max_execution_time && $time_limit > $max_execution_time ) {
+		return @set_time_limit( $time_limit ); // phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
+	}
+	return null;
+}
+
+/**
+ * Does the actual PDF preview regenerate.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bp_document_pdf_previews( $ids, $check_mime_type = false ) {
```

Comment:
@chetansatasiya  Add params in doc block.

Reply:
done